.circleci/config.yml

```diff
@@ -0,0 +1,110 @@
+version: 2
+jobs:
+  test:
+    working_directory: ~/tikv-ci
+    machine: true
+    environment:
+      # CircleCI does not support env variable interpolation, set those later
+      # Variables used by cargo
+      RUST_TEST_THREADS: "1"
+      RUST_BACKTRACE: "1"
+      RUSTFLAGS: "-Dwarnings"
+      FAIL_POINT: "1"
+    steps:
+      - run:
+          name: Updating PATH and Define Environment Variable at Runtime
+          # Define variables that need interpolation
+          command: |
+            echo 'export PATH=$HOME/.local/cmake/bin:$HOME/.cargo/bin:$HOME/.local/bin:$PATH' >> $BASH_ENV
+            echo 'export LD_RUN_PATH=$LD_RUN_PATH:$HOME/.local/lib' >> $BASH_ENV
+            echo 'export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$HOME/.local/lib' >> $BASH_ENV
+            echo 'export LDFLAGS=-L$HOME/.local/lib' >> $BASH_ENV
+            echo 'export CPPFLAGS=-I$HOME/.local/include' >> $BASH_ENV
+            echo 'export CXXFLAGS=-I$HOME/.local/include' >> $BASH_ENV
+            echo 'export LOCAL_PREFIX=$HOME/.local' >> $BASH_ENV
+            echo 'export LD_RUN_PATH=$LD_RUN_PATH:$HOME/.local/lib' >> $BASH_ENV
+            echo 'export LD_RUN_PATH=$LD_RUN_PATH:$HOME/.local/lib' >> $BASH_ENV
+            source $BASH_ENV
+      - checkout
+      - run: 
+          name: Installing Build Dependencies
+          command: |
+            sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y;
+            sudo apt-get update -y;
+            sudo apt-get install -y g++-4.8 curl libdw-dev libelf-dev elfutils binutils-dev
+      - restore_cache:
+          keys: 
+            - v1-local-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
+            - v1-local-{{ .Branch }}
+            - v1-local
+      - restore_cache:
+          keys:
+            - v1-cargo-{{ .Branch }}-{{ checksum "Cargo.lock" }}
+            - v1-cargo-{{ .Branch }}
+            - v1-cargo
+
+      - run: 
+          name: Setting up Rust and Other Build Environment
```

Comment:
"Setting up the build environment" or "Setting up Rust and the rest of the build environment"

Reply:
Thank you advice!